### PR TITLE
Reconstruct OCI part, split it into individual crate, and add tests

### LINKED ISSUE #208: Separate btfhub and oci into Separate Libraries
Currently, the btfhub and oci libraries are bundled together in ecli and ecc.
Separating btfhub and oci into separate libraries can help reduce the size of the overall package.

### LINKED ISSUE #267: [FEATURE] Clean & rewrite & fixbugs of the OCI related part
Related issue:
- https://github.com/eunomia-bpf/eunomia-bpf/issues/208

Currently, the OCI-related part (such as running OCI images) won't work due to many bugs introduced when being reconstructed to Rust. We should try to fix the bugs (if possible, otherwise rewrite), write tests, and split that into an individual library.

- An unfinished OCI library could be found at https://github.com/eunomia-bpf/simoci
- An obvious bug could be seen at [here](https://github.com/eunomia-bpf/eunomia-bpf/blob/11b66aee6f36f551a2b48d4942143591a8e53e2a/ecli/ecli-lib/src/oci/wasm/mod.rs#L30). i.e that the image address is not a URL obviously, but here we try to parse that with a URL parser.

### PR DESCRIPTION
**Pull request by Officeyutong**
Closes #267

Closes #208

Closes https://github.com/eunomia-bpf/bpf-oci/issues/1

This PR reconstructs the OCI part, removed much redundant stuff, and added corresponding tests.

The updates:
- Removed `login` and `logout` options from `ecli`. They store credentials in a local file thus they are unsafe
- Added options `-i, --prompt` for `ecli push` and `ecli pull`. When enabled, ecli will prompt the user to input credentials.
- Added options `-u, --username` and `-p, --password` to manually specify the credentials in the command line
-  `-i`, `-u & -p` conflicts with each other
- If neither of  `-i`, `-u & -p` is provided, will try to read credentials from docker's configuration (`~/.docker/config.json`) and use them to login into the registry. If unable to read, will login to the registry anonymously.
- A new crate `bpf-oci` was added, and contains functions related to OCI. It will be published to crates.io once finished.

### CODE REVIEW

**Comment by yunwei37** [comment]
suggestions

1. rename ecli-oci to bpf-oci, for general purpose and further maintaining
2. --prompt seems not often occurs, is there any better possible flag names?
2. is it better to make read credentials from docker's configuration the default action? if it's not exists, login anonymously

**Comment by Officeyutong** [comment]
> suggestions
> 
> 1. rename ecli-oci to bpf-oci, for general purpose and further maintaining
> 2. --prompt seems not often occurs, is there any better possible flag names?
> 3. is it better to make read credentials from docker's configuration the default action? if it's not exists, login anonymously

1 & 3 is done now. 2 has been discussed to remain unchanged

File: bpf-oci/assets/docker-config-test.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "auths": {
+        "ghcr.io": {
+            "auth": "YWFhOmJiYg=="
```
Comment (yunwei37) [inline bpf-oci/assets/docker-config-test.json:4]: is it hard coded the auth here?

File: bpf-oci/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
+//!  SPDX-License-Identifier: MIT
```
Comment (yunwei37) [inline bpf-oci/src/lib.rs:1]: separate the bpf-oci to another repo?

File: bpf-oci/assets/docker-config-test.json (hunk 1 above)
Comment (Officeyutong) [reply bpf-oci/assets/docker-config-test.json:4]: This file if for testing. The authencatin string is aaa:bbb

File: bpf-oci/src/lib.rs (hunk 2 above)
Comment (Officeyutong) [reply bpf-oci/src/lib.rs]: Done now. It could be found at https://github.com/eunomia-bpf/bpf-oci

**Review by yunwei37 — APPROVED** [review]

**Pull request merged**